ultradns/zone.go

```diff
@@ -0,0 +1,234 @@
+	TransferStatusDetails    *TransferStatusDetails `json:"transferStatusDetails,omitempty"`
+	NotificationEmailAddress string                 `json:"notificationEmailAddress,omitempty"`
+
+	//Alias Zone Response
+	OriginalZoneName string `json:"originalZoneName,omitempty"`
+}
+
+type ZoneListResponse struct {
+	QueryInfo  *QueryInfo      `json:"queryInfo"`
+	ResultInfo *ResultInfo     `json:"resultInfo"`
+	Zones      []*ZoneResponse `json:"zones"`
+}
+
+//create zone
+func (c *Client) CreateZone(zone Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	res, err := c.Do("POST", "zones", zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, fmt.Errorf("error while creating a zone (%v) - %s", zone.Properties.Name, errDataList[0])
+	}
+
+	if res.StatusCode == 202 {
+		taskId := res.Header.Get("X-Task-Id")
+		er := c.TaskWait(taskId, 5, 10)
+
+		if er != nil {
+			return res, er
+		}
+	}
+
+	return res, nil
+}
+
+//read zone
+func (c *Client) ReadZone(zoneName string) (*http.Response, *ZoneResponse, error) {
+	target := Target(&ZoneResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("GET", "zones/"+zoneName, nil, target)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, nil, fmt.Errorf("error while reading a zone (%v) - %s", zoneName, errDataList[0])
+	}
+	zoneResponse := target.Data.(*ZoneResponse)
+
+	return res, zoneResponse, nil
+}
+
+//update zone
+func (c *Client) UpdateZone(zoneName string, zone Zone) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("PUT", "zones/"+zoneName, zone, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, fmt.Errorf("error while updating a zone (%v) - %s", zoneName, errDataList[0])
+	}
+
+	return res, nil
+}
+
+//delete zone
+func (c *Client) DeleteZone(zoneName string) (*http.Response, error) {
+	target := Target(&SuccessResponse{})
+	zoneName = strings.Replace(zoneName, "/", "%2F", 1)
+	res, err := c.Do("DELETE", "zones/"+zoneName, nil, target)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		errDataList := target.Error
+		return res, fmt.Errorf("error while Deleting a zone (%v) - %s", zoneName, errDataList[0])
+	}
+
+	return res, nil
+}
+
+//list Zones
+func (c *Client) ListZone(query string) (*http.Response, *ZoneListResponse, error) {
+	target := Target(&ZoneListResponse{})
+	query = strings.Replace(query, "/", "%2F", 1)
+	res, err := c.Do("GET", "zones/"+query, nil, target)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode > 299 {
```

Comment:
This check is & will be used at many places, I think it would be better to have a generic method to wrap this if block and use it from everywhere.